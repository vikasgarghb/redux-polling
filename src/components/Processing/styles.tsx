import styled, { keyframes } from 'styled-components';

const rotate360 = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`;

interface ILoadingStyled {
  theme?: any;
}

export const ProcessingStyled = styled.div`
  color: #4e5164;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 0.9rem;
  svg {
    animation: ${rotate360} 0.8s steps(12, end) infinite;
    transform: translateZ(0);
    background: transparent;
    margin-top: -3px;
    width: 20px;
    height: 20px;
    display: inline-block;
    vertical-align: middle;
    margin-right: 5px;
  }
`;
