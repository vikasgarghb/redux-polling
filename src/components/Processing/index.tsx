import { ProcessingStyled } from './styles';
import * as React from 'react';
import ProcessorIcon from '../ProcessorIcon';

interface IProcessor {
  title?: string;
}

const Processing = (props: IProcessor) => (
  <ProcessingStyled {...props}>
    <ProcessorIcon />
    {props.title}
  </ProcessingStyled>
);

export default Processing;
