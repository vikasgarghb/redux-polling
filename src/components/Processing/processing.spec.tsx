import React from 'react';
import { shallow } from 'enzyme';

import Processing from '.';

describe('Processing', () => {
  const wrapper = shallow(<Processing title={'processing'} />);
  it('Should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
