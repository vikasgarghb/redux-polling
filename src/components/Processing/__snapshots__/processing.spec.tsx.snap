// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Processing Should match snapshot 1`] = `
<styled.div
  title="processing"
>
  <ProcessorIcon />
  processing
</styled.div>
`;
